opentelemetry-api-shaded-for-instrumenting/build.gradle.kts

```diff
@@ -2,18 +2,46 @@ plugins {
   id("com.github.johnrengelman.shadow")
 
   id("otel.java-conventions")
+  id("otel.publish-conventions")
+}
+
+description = "opentelemetry-api shaded for internal javaagent usage"
+group = "io.opentelemetry.javaagent"
```

Comment:
any interest in `io.opentelemetry.javaagent.build` or something similar? `agent-for-testing` could go there too. `testing-common` could go in `io.opentelemetry.instrumentation.build` or similar. do we have anything else to justify new sub-groupIds?

Reply:
I don't think we need separate groups just for these artifacts that end users consume. One group for all possible artifacts library ones will (testing is part of the deal) and one for agent.

A separate one for this sort of implementation detail could make sense but is probably overkill.